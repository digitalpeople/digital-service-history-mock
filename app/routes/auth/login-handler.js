import { compare } from '../../utils/hash';
import createToken from '../../utils/token';
import {
  mockUsername,
  mockPassword,
} from '../../../data/mock-credentials';

export const postLoginHandler = async (request, response) => {
  const {
    username,
    password,
  } = request.body;

  const usernameMatch = username === mockUsername;
  const passwordMatch = await compare(password, await mockPassword);

  if (usernameMatch && passwordMatch) {
    const token = createToken({ username });

    response.send({
      token,
    });
  } else {
    response.sendStatus(401);
  }
};
