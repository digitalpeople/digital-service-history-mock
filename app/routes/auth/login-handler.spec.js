import { postLoginHandler } from './login-handler';
import createToken from '../../utils/token';

jest.mock('../../utils/token', () => jest.fn(() => 'mock-token'));

describe('postLoginHandler', () => {
  it('should call reponse.send with token', async () => {
    expect.assertions(2);

    const request = {
      body: {
        username: 'test',
        password: '12345',
      },
    };
    const response = {
      send: jest.fn(),
    };

    await postLoginHandler(request, response);

    expect(response.send.mock.calls[0]).toMatchSnapshot();
    expect(createToken.mock.calls[0]).toMatchSnapshot();
  });

  it('should call reponse.sendStatus', async () => {
    expect.assertions(1);

    const request = {
      body: {
        username: 'test',
        password: 'invalid-password',
      },
    };
    const response = {
      sendStatus: jest.fn(),
    };

    await postLoginHandler(request, response);

    expect(response.sendStatus.mock.calls[0]).toMatchSnapshot();
  });
});
