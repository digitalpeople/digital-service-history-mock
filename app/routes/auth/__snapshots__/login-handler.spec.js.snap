// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postLoginHandler should call reponse.send with token 1`] = `
Array [
  Object {
    "token": "mock-token",
  },
]
`;

exports[`postLoginHandler should call reponse.send with token 2`] = `
Array [
  Object {
    "username": "test",
  },
]
`;

exports[`postLoginHandler should call reponse.sendStatus 1`] = `
Array [
  401,
]
`;
